tools/barista/src/main.ts

```diff
@@ -64,9 +82,168 @@ async function buildPages(): Promise<void[]> {
   return Promise.all(files);
 }
 
+function getOverviewSectionItem(
+  filecontent: any,
+  category: string,
+  filepath: string,
+): BaOverviewPageSectionItem {
+  filecontent.properties =
+    filecontent.properties && filecontent.properties.length > 0
+      ? [...filecontent.properties]
+      : [];
+  if (highlightedItems.includes(filecontent.title)) {
+    filecontent.properties.push('favorite');
+  }
+
+  return {
+    title: filecontent.title,
+    identifier:
+      filecontent.title && filecontent.title.length > 1
+        ? filecontent.title[0] + filecontent.title[1]
+        : 'Id',
+    description: filecontent.description,
+    category: category,
+    link: filepath,
+    badge: filecontent.properties,
+    order: filecontent.order,
+  };
+}
+
+/** Builds overview pages */
+async function buildOverviewPages(): Promise<void[]> {
```

Comment:
Should we move this function in a separate file to keep the main file a bit more simple?